src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDAOImpl.java

```diff
@@ -360,4 +366,13 @@ private <T> T execute(String sql, SQLExecution<T> thunk) {
             throw new ShadowManagerDataException(e);
         }
     }
+
+    private <T> T executeWriteOperation(String sql, SQLExecution<T> thunk) {
+        try {
+            return dbWriteThreadPool.submit(() -> execute(sql, thunk)).get(10, TimeUnit.SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)